### PR TITLE
Add shell completion

### PR DESCRIPTION
**Pull request by thomas-riccardi**
Test with:
eval "$(register-python-argcomplete dmake)"

Also add `dmake completion` command and setup in install.sh.

### CODE REVIEW

File: dmake/dmake [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env python
+# PYTHON_ARGCOMPLETE_OK
```
Comment (Alexis-Jacob) [inline dmake/dmake]: ?

Comment (thomas-riccardi) [reply dmake/dmake]: That's from their doc..

**Comment by Alexis-Jacob** [comment]
It's working correctly, however it's strange that the options are mixed between the optional and the mandatory 
```bash
alexis@laptop:~/projects/tests$ dmake shell --no-dependencies 
-b                            celery-on-premises            detection                     gui                           --no-dependencies             thoth-worker-on-premises      web-on-premises 
--branch                      --command                     detection-on-premises         gui-on-premises               -s                            training-worker               
-c                            -d                            e2e                           -h                            --standalone                  training-worker-on-premises   
celery                        --dependencies                e2e-on-premises               --help                        thoth-worker                  web        
```

**Comment by thomas-riccardi** [comment]
> It's working correctly, however it's strange that the options are mixed between the optional and the mandatory
> 
> ```shell
> alexis@laptop:~/projects/tests$ dmake shell --no-dependencies 
> -b                            celery-on-premises            detection                     gui                           --no-dependencies             thoth-worker-on-premises      web-on-premises 
> --branch                      --command                     detection-on-premises         gui-on-premises               -s                            training-worker               
> -c                            -d                            e2e                           -h                            --standalone                  training-worker-on-premises   
> celery                        --dependencies                e2e-on-premises               --help                        thoth-worker                  web        
> ```

This is a MVP, we will improve the completion results and the speed (if possible) later.

**Pull request merged**